base/show.jl

```diff
@@ -12,6 +12,15 @@ end
 
 show(io::IO, ::MIME"text/plain", r::AbstractRange) = show(io, r) # always use the compact form for printing ranges
 
+function show(io::IO, ::MIME"text/plain", r::UnitRange)
+    show(io, first(r))
+    print(io, ':')
+    show(io, last(r))
```

Comment:
```suggestion
    show(io, r)
```
The implementation here should match the 2-arg implementation (so might as well call the 2-arg method). 

While @nsako's suggestion is reasonable, it shouldn't only be applied to the 3-arg and not the 2-arg method because that would cause inconsistencies like this:

```julia-repl
julia> io = IOContext(stdout, :typeinfo=>Bool);

julia> show(io, MIME("text/plain"), false:true)
0:1
julia> show(io, MIME("text/plain"), false:true:true)
false:true:true
```

I would recommend attempting that performance improvement in a separate pull request due to its semantic impact. Calling `show(io, r)` here means that any performance improvements to `show(io, r)` will also apply here.